### PR TITLE
Update beforePrint to fix IE print cutoffs

### LINKED ISSUE #3983: IE: Incorrect printing in Internet Explorer
Pdf.js demo is incorrectly printed in IE9-IE11.
OS: Windows 7 64 bit, Windows 8
Here we go:
http://mozilla.github.io/pdf.js/web/viewer.html
Try to print a page in IE 9 with default settings.
Result is horrible:
pdfjs_portrait

This problem can be solved, if we call "Page Setup" dialog in IE 9 and clear "Enable Shrink-to-Fit" checkbox. By the way, it must be stated in wiki somethere.
Also, when we print a pdf with landscape orientation, pdfjs is not smart enough to rotate canvases in IE (Chrome printing is OK), user must set orientation by hand.

But even with  "Enable Shrink-to-Fit" cleared, printing is incorrect: we have overlapped pages.
Example:
pdfjs_overlap
v0:= ld state[748] is printed twice, and subsequent pages are even more overlapped.

It seems to me, we can fix this problem with style:
<style>
#printContainer div{
    overflow: hidden;
}
</style>

But there is still a problem, we have an empty page printed after all pdf's pages. We could not solve this

### LINKED ISSUE #6345: Print on Chrome 45 prints extra blank pages
This was fixed by #5857 but break again on Chrome 45.0.2454.26 beta (64-bit) and OS X 10.10.4.
It is working on Chrome 44.
I tried many PDFs (including compressed.tracemonkey-pldi-09.pdf) and all of them have extra blank page before actual content page.

### PR DESCRIPTION
**Pull request by FesterCluck**
Fixes #3983.
Fixes #6345.

### CODE REVIEW

File: web/pdf_page_view.js [hunk 1]
```diff
@@ -490,10 +490,11 @@ var PDFPageView = (function PDFPageViewClosure() {
     beforePrint: function PDFPageView_beforePrint() {
       var pdfPage = this.pdfPage;
 
-      var viewport = pdfPage.getViewport(1);
+      var viewport = pdfPage.getViewport(2);
       // Use the same hack we use for high dpi displays for printing to get
       // better output until bug 811002 is fixed in FF.
-      var PRINT_OUTPUT_SCALE = 2;
+      // In this fork I have not tested for regressions here
```
Comment (timvandermeij) [inline web/pdf_page_view.js]: Remove this line as it is not useful for the code itself. Before we land this, we must have tested for regressions anyway.

**Comment by timvandermeij** [comment]
@FesterCluck Could you squash the commits? See https://github.com/mozilla/pdf.js/wiki/Squashing-Commits on how to do that easily.

@Rob--W Do you have ideas how we can test this properly to make sure that no regressions arise? I'm not that familiar with the printing code. Of course we shall do some cross-browser tests.

**Comment by Rob--W** [comment]
@timvandermeij I think that a manual test plan would be best here.

1. Open PDF in PDF.js.
2. Click on the Print button.
3. Look at the print preview if available (otherwise print to a (PDF) file), and compare the result with what's rendered in the PDF.js PDF Viewer.
4. If there are no significant deviations, consider the test successful.

**Comment by FesterCluck** [comment]
@timvandermeij First time I've done a squash, could you double check I've done this right?

**Comment by timvandermeij** [comment]
@FesterCluck Yes, that looks good. Could you also address the comment I added above? Once you have done that in a new commit, please re-squash the commits to make review easier for us. Thanks!

**Comment by FesterCluck** [comment]
@timvandermeij comment removed.

**Comment by timvandermeij** [comment]
@FesterCluck Looks good, thanks.

**Comment by Hengjie** [comment]
This may affect https://github.com/mozilla/pdf.js/issues/5718 as that PR also moves away from `pt` measurements. I can test this later unless someone does it before me.

**Comment by CodingFabian** [comment]
looks simpler than #5718 for me. i can support testing this. can we make a preview build?

**Comment by timvandermeij** [comment]
Yes, if you would help out with testing this, that would be great!

/botio-linux preview

**Comment by Hengjie** [comment]
Haven't noticed any regressions on v43, v44, Firefox 39.0.

File: web/pdf_page_view.js [hunk 2]
```diff
@@ -496,10 +496,10 @@ var PDFPageView = (function PDFPageViewClosure() {
     beforePrint: function PDFPageView_beforePrint() {
       var pdfPage = this.pdfPage;
 
-      var viewport = pdfPage.getViewport(1);
+      var viewport = pdfPage.getViewport(2);
       // Use the same hack we use for high dpi displays for printing to get
       // better output until bug 811002 is fixed in FF.
-      var PRINT_OUTPUT_SCALE = 2;
+      var PRINT_OUTPUT_SCALE = 1;
```
Comment (brendandahl) [inline web/pdf_page_view.js]: If we're changing this to 1 we should just remove everything related to PRINT_OUTPUT_SCALE since it now does nothing.

**Comment by brendandahl** [comment]
I'm leery of moving away from pts as I don't think it will give the correct scale. PDF units are given in pts (1/72in), so it makes sense to have the canvas the correct size. We'd need to verify these changes don't affect the scale when printed at 100%. 

http://www.vendian.org/mncharity/dir3/paper_rulers/

**Comment by StoicLoofah** [comment]
Just wanted to chime in and say that I also had to apply this change locally to get printing working in IE9 - IE11, though it appears to not be necessary in Edge. The issue I saw was only getting the top half (or less) of each page printed. I didn't see any regressions in FF 46. I can't speak to the pt/px difference or interactions with other issues, but it worked for me in practice.

**Comment by timvandermeij** [comment]
Closing as superseded by #7487 because that patch fixes some (potential) issues with this patch.

**Pull request closed without merging**